src/zjs_fs.c

```diff
@@ -524,9 +544,16 @@ static jerry_value_t zjs_write(const jerry_value_t function_obj,
         return zjs_error("offset + length overflows buffer");
     }
 
-    // if a position was specified, seek to it before writing
-    if (fs_seek(&handle->fp, (uint32_t)position, SEEK_SET) != 0) {
-        return zjs_error("error seeking to position\n");
+    if (handle->mode == MODE_A || handle->mode == MODE_A_PLUS) {
+        // if in append mode, seek to end (ignoring position parameter)
+        if (fs_seek(&handle->fp, 0, SEEK_END) != 0) {
+            return zjs_error("error seeking start");
+        }
+    } else {
+        // if a position was specified, seek to it before writing
+        if (fs_seek(&handle->fp, (uint32_t)position, SEEK_SET) != 0) {
```

Comment:
Again you should allow position to be null above, and do this seek only if it's not null.

Yuck - apparently you're right in the arg handling code above that position is *optional* for write, even though it was not optional for read. Ew. But it can still be null.